### PR TITLE
feat: add ability to compare versions

### PR DESCRIPTION
**Pull request by mitsuhiko**
This adds the ability to compare versions to each other.

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -537,3 +585,38 @@ fn test_environment_validation() {
...
+    assert_eq!(ver!("1.0"), ver!("1.0.0"));
+    assert_eq!(ver!("1.0dev"), ver!("1.0.0-dev"));
+    assert_eq!(ver!("1.0dev"), ver!("1.0.0.0-dev"));
+    assert_ne!(ver!("1.0dev"), ver!("1.0.0.0-dev1"));
+
+    assert!(ver!("1.0dev") >= ver!("1.0.0.0-dev"));
+    assert!(ver!("1.0dev") < ver!("1.0.0.0"));
+
+    assert!(ver!("1.0.0") > ver!("1.0.0-rc1"));
+    assert!(ver!("1.0.0") >= ver!("1.0.0-rc1"));
```
Comment (ahmedetefy) [inline src/parser.rs:608]: I don't understand why line 608 would pass, it should fail no?

File: src/parser.rs [hunk 2]
```diff
@@ -537,3 +585,38 @@ fn test_environment_validation() {
...
+    assert!(ver!("1.0") < ver!("2.0"));
+    assert!(ver!("1.1.0") < ver!("10.0"));
+    assert!(ver!("1.1.0") < ver!("1.1.1"));
+    assert!(ver!("1.1.0.1") < ver!("1.1.0.2"));
+    assert!(ver!("1.1.0.1") > ver!("1.1.0.0"));
+    assert!(ver!("1.1.0.1") > ver!("1.0.0.0"));
+    assert!(ver!("1.1.0.1") > ver!("1.0.42.0"));
+
+    assert!(ver!("1.0+abcd") < ver!("1.0+abcde"));
+}
```
Comment (ahmedetefy) [inline src/parser.rs:622]: Nit: I would add these just to confirm behavior

```suggestion
assert!(ver!("1.0.0-alpha.beta") < ver!("1.0.0-beta"));
assert!(ver!("1.0.0-beta") < ver!("1.0.0-beta.2"));
assert!(ver!("1.0.0-alpha.2") < ver!("1.0.0-alpha.11"));
assert!(ver!("1.0.0-beta.11") < ver!("1.0.0-rc.1"));
}
```

**Review by ahmedetefy — CHANGES_REQUESTED** [review]

File: src/parser.rs (hunk 1 above)
Comment (mitsuhiko) [reply src/parser.rs:608]: why should it fail? `1.0.0` is greater or equal to the pre-release version `1.0.0-rc1`.

Comment (ahmedetefy) [reply src/parser.rs:608]: Shouldn't it be just greater than?

**Review by ahmedetefy — APPROVED** [review]

File: src/parser.rs (hunk 1 above)
Comment (mitsuhiko) [reply src/parser.rs:608]: it's both :)

**Pull request merged**